internal/server/iigointernal/executive.go

```diff
@@ -0,0 +1,148 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/voting"
+)
+
+type executive struct {
+	ID               shared.ClientID
+	clientPresident  roles.President
+	budget           shared.Resources
+	speakerSalary    shared.Resources
+	RulesProposals   []string
+	ResourceRequests map[shared.ClientID]shared.Resources
+}
+
+// returnSpeakerSalary returns the salary to the common pool.
+func (e *executive) returnSpeakerSalary() shared.Resources {
+	x := e.speakerSalary
+	e.speakerSalary = 0
+	return x
+}
```

Comment:
returns to common pool?

Reply:
I believe this is a hot-fix for the bug that was about resources being taken but not returned in the case that IIGO runs. For post MVP this system needs to be reworked and expanded because firstly the order of events should be (decision)->(try withdraw from CP->add to budget) -> (add budget leftovers to island's resources) not (try withdraw from CP)->(decision)->(add to budget) -> (add budget leftovers to island's resources). Moreover, #124 will change how we interact with the CP anyways. Hence this is a temporary thing whilst it is assumed that whatever an island is permitted to pay another island for its role is actually what the island pays and that all IIGO actions always happen (unless all of IIGO does not run) thus using up all the budget